app/presenters/property_presenter.rb

```diff
@@ -98,38 +79,45 @@ def render_table
     ).concat(output)
   end
 
-  def render_tab_content(values)
-    content_tag(:div, class: 'tab-content') do
-      values.map do |entry|
-        content_tag(:div, class: 'tab-pane', id: "#{entry[:protocol]}-#{entry[:port]}-tab") do
-          render_scripts_table(entry)
+  def merge_rows(rows)
+    rows.group_by{|e| [e[:port], e[:protocol]]}.map do |_, a|
+      a.reduce do |memo, row|
+        memo.merge(row) do |_key, oldvalue, newvalue|
+          if oldvalue == newvalue
+            oldvalue
+          elsif oldvalue.respond_to?(:concat)
+            backupvalue = oldvalue.clone
+            backupvalue.concat(content_tag(:br)).concat(newvalue).html_safe
+          elsif oldvalue.respond_to?(:merge)
+            oldvalue.merge(newvalue)
+          end
         end
-      end.join.html_safe
+      end
     end
   end
 
-  def render_tab_ul(values)
-    content_tag :ul, class: 'nav nav-tabs' do
-      values.map do |entry|
-        content_tag :li do
-          content_tag(:a,
-            "%s/%d" % [entry[:protocol], entry[:port]],
-            data: { toggle: :tab },
-            href: "#%s-%d-tab" % [entry[:protocol], entry[:port]]
-          )
-        end
-      end.join.html_safe
-    end
+  def services_table_columns
+    NodeProperties::SERVICE_KEYS
   end
 
-  def render_tabs(entries)
-    values = entries.select{ |entry| entry.key?(:scripts) && entry[:scripts].any? }
-    return if values.empty?
-
-    content_tag(:h4, 'NSE script output') +
-    content_tag(:div, class: 'tabbable tabs-left', id: 'scripts-tabs') do
-      concat(render_tab_ul(values)).
-      concat(render_tab_content(values))
+  # Prepare info to be displayed in services table
+  # converts
+  # [ {"port" => 80, "name" => "foo"},
+  #   {"port" => 22, "name" => "bar"},
+  #   {"port" => 80, "name" => "baz"}, ... ]
+  # into
+  # [ {"port" => 22, "name" => "baz"},
+  #   {"port" => 80, "name" => "foo<br></br>bar"}, ... ]
+  # Id sort_by_port is true it merges duplicated rows by [port/protocol],
+  # adding duplicted values with <br/><br/>
+  # It always filters the keys to those available in `services_table_columns`
+  def table_info(values:, sort_by_port:)
```

Comment:
I don't think it's this layer's Is it the view layer's responsibility to do this de-dup (e.g. to know what properties should be considered equivalent and hence merged?)

I'd argue is the `NodeProperties` concern responsibility, maybe via a special accessor.

Otherwise each view layer needs to replicate the de-dup code (I'm thinking Word rendering). I'm aware that each layer needs to have some code to go from properties to the table representation of the view format (HTML vs Word), but that's view-specific details. De-dupping is a higher level concept.